service/service_test.go

```diff
@@ -498,6 +515,75 @@ func processingOfPaymentKafkaMessageCreatesReconciliationRecords(
 	})
 }
 
+// processingOfRefundKafkaMessageCreatesReconciliationRecords asserts that a refund of the class specified
+// will result in a call to get the payment details and the creation of refund reconciliation records.
+func processingOfRefundKafkaMessageCreatesReconciliationRecords(
+	ctrl *gomock.Controller,
+	productMap *config.ProductMap,
+	classOfPayment string) {
+
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	c := make(chan os.Signal)
+
+	mockPayment := payment.NewMockFetcher(ctrl)
+	mockTransformer := transformer.NewMockTransformer(ctrl)
+	mockDao := dao.NewMockDAO(ctrl)
+
+	svc := createMockService(productMap, mockPayment, mockTransformer, mockDao)
+
+	Convey("Given a message is readily available for the service to consume", func() {
+
+		svc.Consumer = createMockConsumerWithMessage(paymentResourceID, refundID)
+
+		Convey("When the payment corresponding to the message is fetched successfully", func() {
+
+			cost := data.Cost{
+				ClassOfPayment: []string{classOfPayment},
+			}
+
+			pr := data.PaymentResponse{
+				CompanyNumber: "123456",
+				Costs:         []data.Cost{cost},
+				Refunds: []data.RefundResource{{
+					RefundId:          refundID,
+					CreatedAt:         "",
+					Amount:            0,
+					Status:            "success",
```

Comment:
what about a test for a non-successful refund? Or a payment object with mutliple refunds?